pkg/filewatch/filewatch_test.go

```diff
@@ -0,0 +1,149 @@
+package filewatch
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Filewatch", func() {
+	Context("watching file", func() {
+		var (
+			callback     func()
+			tempFileName string
+		)
+
+		BeforeEach(func() {
+			tmpDir := GinkgoT().TempDir()
+			tempFileName = filepath.Join(tmpDir, "test-file")
+			Expect(os.WriteFile(tempFileName, []byte("test content"), 0777)).To(Succeed())
+
+			callback = func() {
+				defer GinkgoRecover()
+				panic("callback was not set")
+			}
+
+			testWatch := New()
+			err := testWatch.Add(tempFileName, func() { callback() })
+			Expect(err).ToNot(HaveOccurred())
+
+			done := make(chan struct{})
+			DeferCleanup(func() {
+				close(done)
+			})
+
+			go func() {
+				defer GinkgoRecover()
+				Expect(testWatch.Run(done)).To(Succeed())
+			}()
+
+			// Wait for a short time to let the watch goroutine run
+			runtime.Gosched()
+			Eventually(testWatch.IsRunning, time.Second, 50*time.Millisecond).Should(BeTrue())
+		})
+
+		It("should call callback on file change", func() {
+			called := atomic.Bool{}
+			callback = func() { called.Store(true) }
+
+			Expect(os.WriteFile(tempFileName, []byte("different content"), 0777)).To(Succeed())
+
+			Eventually(called.Load, time.Second, 50*time.Millisecond).Should(BeTrue())
+		})
+
+		It("should call callback on file deletion", func() {
+			called := atomic.Bool{}
+			callback = func() { called.Store(true) }
+
+			Expect(os.Remove(tempFileName)).ToNot(HaveOccurred())
+
+			Eventually(called.Load, time.Second, 50*time.Millisecond).Should(BeTrue())
+		})
+	})
+
+	Context("watching directory", func() {
+		var (
+			callback    func()
+			tempDirName string
+		)
+
+		BeforeEach(func() {
```

Comment:
Can you dedup this `BeforeEach` and the one in L22?

Reply:
done